<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NodeTreePanel</class>
 <widget class="QDockWidget" name="NodeTreePanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>336</width>
    <height>566</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Node Tree</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>6</number>
          </property>
          <item>
           <widget class="QToolButton" name="m_newNodeButton">
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>...</string>
            </property>
            <property name="icon">
             <iconset resource="../../../Resource/App.qrc">
              <normaloff>:/icon/Icon/project/add.png</normaloff>:/icon/Icon/project/add.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QT_UI::QDragDropTreeWidget" name="m_nodeTreeWidget">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::EditKeyPressed</set>
          </property>
          <property name="dragEnabled">
           <bool>true</bool>
          </property>
          <property name="dragDropMode">
           <enum>QAbstractItemView::InternalMove</enum>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="headerHidden">
           <bool>true</bool>
          </property>
          <property name="columnCount">
           <number>2</number>
          </property>
          <attribute name="headerVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="headerDefaultSectionSize">
           <number>200</number>
          </attribute>
          <attribute name="headerStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string notr="true">1</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string notr="true">2</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="tabWidget">
       <property name="tabPosition">
        <enum>QTabWidget::South</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Triangular</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="elideMode">
        <enum>Qt::ElideMiddle</enum>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Property</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTreeView" name="m_propertyTreeView">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="editTriggers">
            <set>QAbstractItemView::AllEditTriggers</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Signal</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTreeWidget" name="m_signalTreeWidget">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="headerHidden">
            <bool>true</bool>
           </property>
           <property name="columnCount">
            <number>1</number>
           </property>
           <column>
            <property name="text">
             <string notr="true">1</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="m_actionDeleteNode">
   <property name="icon">
    <iconset resource="../../../Resource/App.qrc">
     <normaloff>:/icon/Icon/minus.png</normaloff>:/icon/Icon/minus.png</iconset>
   </property>
   <property name="text">
    <string>Delete Node(s)</string>
   </property>
   <property name="iconText">
    <string>Delete Node(s)</string>
   </property>
  </action>
  <action name="m_actionRenameNode">
   <property name="text">
    <string>Rename</string>
   </property>
  </action>
  <action name="m_actionImportGltfScene">
   <property name="icon">
    <iconset resource="../../../Resource/App.qrc">
     <normaloff>:/icon/node/gltfscene.png</normaloff>:/icon/node/gltfscene.png</iconset>
   </property>
   <property name="text">
    <string>Import glTF Scene</string>
   </property>
  </action>
  <action name="m_actionAddNode">
   <property name="icon">
    <iconset resource="../../../Resource/App.qrc">
     <normaloff>:/icon/node/node.png</normaloff>:/icon/node/node.png</iconset>
   </property>
   <property name="text">
    <string>Add Child Node</string>
   </property>
  </action>
  <action name="m_actionAddChildScene">
   <property name="text">
    <string>Instance Child Scene</string>
   </property>
   <property name="toolTip">
    <string>Instance Child Scene</string>
   </property>
  </action>
  <action name="m_actionChangeType">
   <property name="text">
    <string>Change Type</string>
   </property>
  </action>
  <action name="m_actionSaveBranchasScene">
   <property name="text">
    <string>Save Branch As Scene</string>
   </property>
  </action>
  <action name="m_actionDiscardInstancing">
   <property name="text">
    <string>Discard Instancing</string>
   </property>
  </action>
  <action name="m_actionDuplicateNode">
   <property name="text">
    <string>Duplicate</string>
   </property>
  </action>
  <action name="m_actionConnectSlot">
   <property name="text">
    <string>Connect Slot</string>
   </property>
   <property name="toolTip">
    <string>Connect Signal Slot</string>
   </property>
  </action>
  <action name="m_actionDisconnect">
   <property name="text">
    <string>Disconnect</string>
   </property>
   <property name="toolTip">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="m_actionDisconnectAll">
   <property name="text">
    <string>Disconnect All</string>
   </property>
   <property name="toolTip">
    <string>Disconnect All</string>
   </property>
  </action>
  <action name="m_actionConnectEdit">
   <property name="text">
    <string>Edit</string>
   </property>
   <property name="toolTip">
    <string>Edit Connection</string>
   </property>
  </action>
  <action name="m_actionGoToMethod">
   <property name="text">
    <string>Go to Method</string>
   </property>
   <property name="toolTip">
    <string>Go To Method</string>
   </property>
  </action>
  <action name="m_actionPropertyReference">
   <property name="text">
    <string>Reference</string>
   </property>
   <property name="toolTip">
    <string>Reference</string>
   </property>
  </action>
  <action name="m_actionDeletePropertyReference">
   <property name="text">
    <string>Delete Reference</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QT_UI::QDragDropTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>QDragDropTreeWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../Resource/App.qrc"/>
 </resources>
 <connections/>
</ui>
